src/IRGraph.purs

```diff
@@ -377,29 +383,28 @@ forUnion_ (IRUnionRep { primitives, arrayType, classRef, mapType }) f = do
 
 mapUnionM :: forall a m. Monad m => (IRType -> m a) -> IRUnionRep -> m (List a)
 mapUnionM f (IRUnionRep { primitives, arrayType, classRef, mapType }) = do
-    let l = L.Nil
-    l <- mapGeneral l mapType IRMap
-    l <- mapGeneral l classRef IRClass
-    l <- mapGeneral l arrayType IRArray    
-    l <- mapPrimitive l irUnion_String IRString
-    l <- mapPrimitive l irUnion_Bool IRBool
-    l <- mapPrimitive l irUnion_Double IRDouble
-    l <- mapPrimitive l irUnion_Integer IRInteger
-    l <- mapPrimitive l irUnion_Null IRNull
-    l <- mapPrimitive l irUnion_Nothing IRNothing
-    pure l
+    pure L.Nil
+        >>= mapGeneral mapType IRMap
+        >>= mapGeneral classRef IRClass
+        >>= mapGeneral arrayType IRArray
+        >>= mapPrimitive irUnion_String IRString
+        >>= mapPrimitive irUnion_Bool IRBool
+        >>= mapPrimitive irUnion_Double IRDouble
+        >>= mapPrimitive irUnion_Integer IRInteger
+        >>= mapPrimitive irUnion_Null IRNull
+        >>= mapPrimitive irUnion_Nothing IRNothing
```

Comment:
This slab of changes was because I was getting a bunch of shadow warnings for `l`.